src/sync.c

```diff
@@ -69,6 +69,12 @@ static uint8_t gray8(float f)
         return 1;
 }
 
+static inline uint8_t demod(float x, float mult)
+{
+    float clamped = fmaxf(fminf(x, 1), -1);
```

Comment:
Clamping the samples like this reduces performance slightly in low-SNR AWGN channels. But it noticeably improves performance in cases where there is adjacent-channel interference.

I believe the issue is that the noise power estimate (which is computed once for the entire sideband) is inaccurate when narrow-band interference corrupts only a small part of the sideband. Perhaps in the future we could switch to more granular noise power estimates.